compiler/rustc_passes/src/check_attr.rs

```diff
@@ -400,6 +402,83 @@ impl CheckAttrVisitor<'tcx> {
         }
     }
 
+    /// Checks if `#[rustc_args_required_const]` is applied to a function and has a valid argument.
+    fn check_rustc_args_required_const(
+        &self,
+        attr: &Attribute,
+        span: &Span,
+        target: Target,
+        item: Option<&Item<'_>>,
+    ) -> bool {
+        match target {
+            Target::Fn | Target::Method(..) | Target::ForeignFn => {
+                let mut invalid_args = vec![];
+                if let Some(meta_item_list) = attr.meta_item_list() {
+                    for meta in meta_item_list {
+                        if let Some(lit) = meta.literal() {
+                            match lit.kind {
+                                LitKind::Int(val, lit_int_type) => {
+                                    if !matches!(lit_int_type, LitIntType::Unsuffixed) {
+                                        invalid_args.push(lit.span);
```

Comment:
afaik we already emit an error for this somewhere else 
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=9e24c5ae1ab82c13c8fabdeb2716dfb3

can you add a test for this so that we don't emit 2 distinct errors in that case?

Reply:
Good catch :+1: